Content.IntegrationTests/Tests/Damageable/GeneralisedDamageableTest.cs

```diff
@@ -0,0 +1,171 @@
+using System.Linq;
+using System.Threading.Tasks;
+using Content.Shared.Damage;
+using Content.Shared.Damage.Components;
+using NUnit.Framework;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Map;
+using Robust.Shared.Prototypes;
+
+namespace Content.IntegrationTests.Tests.Damageable
+{
+    [TestFixture]
+    [TestOf(typeof(DamageableComponent))]
+    public class GeneralisedDamageableTest : ContentIntegrationTest
+    {
+        private const string DamageableEntityId = "TestDamageableEntityId";
+        private const string Group1Id = "TestGroup1";
+        private const string Group2Id = "TestGroup2";
+        private const string Group3Id = "TestGroup3";
+        private const string SharedDamageTypeId = "TestSharedDamage";
+        private const string UnsupportedDamageTypeId = "TestUnsupportedDamage";
+        private string Prototypes = $@"
+- type: damageType
+  id: {SharedDamageTypeId}
+
+- type: damageType
+  id: {UnsupportedDamageTypeId}
+
+- type: damageType
+  id: TestDamage1
+
+- type: damageType
+  id: TestDamage2
+
+- type: damageGroup
+  id: {Group1Id}
+  damageTypes:
+    - {SharedDamageTypeId}
+
+- type: damageGroup
+  id: {Group2Id}
+  damageTypes:
+    - {SharedDamageTypeId}
+    - TestDamage1
+
+- type: damageGroup
+  id: {Group3Id}
+  damageTypes:
+    - {SharedDamageTypeId}
+    - TestDamage2
+    - {UnsupportedDamageTypeId}
+
+# we want to test a container that only partially supports a group:
+- type: damageContainer
+  id: TestPartiallySupported
+  supportedGroups:
+    - {Group2Id}
+  supportedTypes:
+    - TestDamage2
+    - TestDamage1
+# does NOT support type {UnsupportedDamageTypeId}, and thus does not fully support group {Group3Id}
+# TestDamage1 is added twice because it is also in {Group2Id}. This should not cause errors.
+
+# create entities
+- type: entity
+  id: {DamageableEntityId}
+  name: {DamageableEntityId}
+  components:
+  - type: Damageable
+    damageContainer: TestPartiallySupported
+";
+
+        /// <summary>
+        ///     Generalized damageable component tests.
+        /// </summary>
+        /// <remarks>
+        ///     Test scenarios where damage types are members of more than one group, or where a component only supports a subset of a group.
+        /// </remarks>
+        [Test]
+        public async Task TestGeneralizedDamageableComponent()
```

Comment:
whats the diff between this and DamageableTest? can they be merged/put in the same file?

Reply:
I've moved them to be two functions/tests in the same file. They do test slightly different things, so I think its worth keeping them as two different tests. One just looks at the standard damageable component functions, and makes sure they work. The other one specifically checks if there are any issues if a damage type is a member of more than one damage group, or if a damageable component supports a single member of a damage group, without supporting the whole group. If damage specifiers/datasets are added, a lot of that becomes uneccesarry and the test can just be deleted/merged.